File: src/algorithms/third-maximum-number.js
/**
 * Given an integer array nums, returns the third distinct maximum number. 
 * If the third maximum does not exist, returns the maximum number.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number} - Returns the third distinct maximum number from an array
 */
function thirdMax (nums) {
  let distinctNums = new Set();
  let thirdDistinctNum;
  for (let i = 0; i < nums.length; i++) {
    if (!distinctNums.has(nums[i])) {
      distinctNums.add(nums[i]);
    }
  }
  distinctNums = Array.from(distinctNums);
  distinctNums = distinctNums.sort((a, b) => a - b);
  if (distinctNums.length < 3) {
    thirdDistinctNum = distinctNums[distinctNums.length - 1]; 
  } else {
    thirdDistinctNum = distinctNums[distinctNums.length - 3];
  }
  return thirdDistinctNum;
}

File: src/algorithms/third-maximum-number.py
def third_max(nums):
    """Given an integer array nums, return the third distinct maximum number in
    this array. If the third maximum number does not exist, return the maximum
    number.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns the third maximum number in the list.
    """
    nums = list(set(nums))
    nums = sorted(nums)
    if len(nums) >= 3:
        return nums[-3]
    else:
        return nums[-1]
